### PR TITLE
Boemor/splitbutton dependency fix

### PR DESCRIPTION
**Pull request by Boris-Em**
#### Pull request checklist

This PR removes the IconButton dependency from BaseButton that was introduced with SplitButton.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -15,8 +15,10 @@ import { Icon, IIconProps } from '../../Icon';
 import { DirectionalHint } from '../../common/DirectionalHint';
 import { ContextualMenu, IContextualMenuProps } from '../../ContextualMenu';
 import { IButtonProps, IButton } from './Button.Props';
-import { IconButton } from './IconButton/IconButton';
 import { IButtonClassNames, getClassNames } from './BaseButton.classNames';
+import { getStyles as getSplitButtonStyles } from './SplitButton/SplitButton.styles';
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: hmmm I'm confused; why do you need this? Shouldn't the variants just define their own splitbutton styling?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: BaseButton should render the splitbutton content. But the styling for how the split looks may vary from variant to variant.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -352,7 +357,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
         aria-expanded={ this._isExpanded }
         aria-pressed={ this.props.checked }
         aria-describedby={ buttonProps.ariaDescription }
-        className={ css(disabled ? styles.splitButtonContainerDisabled : styles.splitButtonContainer) }
+        className={ css(disabled ? splitButtonStyles.splitButtonContainerDisabled : splitButtonStyles.splitButtonContainer) }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Shouldn't this get the class names from the given styles that are passed into props?

**Comment by Boris-Em** [comment]
Yes @dzearing, I'm revisiting the approach I had taken.

**Comment by Boris-Em** [comment]
Closing this PR as #2208 should address @dzearing's comments.

**Pull request closed without merging**